komorebi/src/window_manager.rs

```diff
@@ -1705,6 +1706,26 @@ impl WindowManager {
         self.update_focused_workspace(self.mouse_follows_focus)
     }
 
+    #[tracing::instrument(skip(self))]
+    pub fn toggle_layout(&mut self) -> Result<()> {
```

Comment:
Instead of this, can you implement a `cycle` method on `DefaultLayout`?

```
impl DefaultLayout {
    pub fn cycle(self) -> Self {
        match self {
            Self::BSP => Self::Columns,
            Self::Columns => Self::Rows,
            Self::Rows => Self::VerticalStack,
            Self::VerticalStack => Self::HorizontalStack,
            Self::HorizontalStack => Self::UltrawideVerticalStack,
            Self::UltrawideVerticalStack => Self::BSP,
        }
    }
}
```

I think this is a bit simpler and avoids unwraps. You should be able to call `cycle()` on `workspace.layout()?` and then `set_layout` with the returned value.

Reply:
Just to add to this; I think we only want the call to `cycle` to go ahead if `Layout` is the `DefaultLayout` variant, otherwise the command should be swallowed silently (ie. the user has explicitly opted in to use their custom layout)

Reply:
Good points, @LGUG2Z - Thanks for the suggestion.

Searched a bit more if we could avoid hard coding layouts in the `cycle` method, but doesn't seem feasible (w/o using any external traits). Went ahead with what you suggested, and also added a comment in the `DefaultLayout` enum suggesting users to register the new layout in the `cycle` method so that we don't miss this.

I can also help add a test later to ensure a new layout is always registered in the `cycle` method (/de-registered if removed).